pkg/apis/operator/v1alpha1/constants.go

```diff
@@ -28,4 +28,8 @@ const (
 	// OperationRotateWorkloadIdentityKeyComplete is a constant for an annotation on a Shoot indicating that the
 	// rotation of the workload identity signing key shall be completed.
 	OperationRotateWorkloadIdentityKeyComplete = "rotate-workload-identity-key-complete"
+
+	// AnnotationKeyExtensionResourcesCleanedUp is the annotation key on the garden resource to mark
+	// that all extension resources like DNSRecords and disposable BackupBuckets have been deleted.
+	AnnotationKeyExtensionResourcesCleanedUp = "operator.gardener.cloud/extension-resources-cleaned-up"
```

Comment:
Instead of working with this special annotation handling, can't we work with the in-memory state of the required map (see [here](https://github.com/gardener/gardener/blob/ddde6b6121857c3a68cbcb88a4638d59930c79e2/pkg/operator/controller/extension/reconciler.go#L61))?

Reply:
Also, we collide a little bit with https://github.com/gardener/gardener/pull/10650 which implements the final deletion procedure. So, if we want to merge this PR first we should try to work with a reduced band-aid only.

Reply:
As already discussed, after #10650 is merged, I will address this point and remove the annotation handling again.